src/p2p_sync.rs

```diff
@@ -61,18 +54,18 @@ where
 
         // listen for response with their public key
         let mut buffer = [0u8; 32];
-        stream.read(&mut buffer)?;
+        stream.read_exact(&mut buffer)?;
 
         // generate encryption key with the private key and their public key
-        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer)?);
+        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer)?)?;
```

Comment:
Since no error is going to be thrown from **public_key_from_bytes**, there is no need to expect an error
```suggestion
        let key = keys.generate_encryption_key(&Keys::public_key_from_bytes(buffer))?;
```